src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -49,7 +56,46 @@ public string ReadOnly
                 get { return null; }
             }
 
+            public ISet<string> NonInstantiatedISet { get; set; } = null!;
+            public IDictionary<string, ISet<string>> NonInstantiatedDictionaryWithISet { get; set; } = null!;
+            public IDictionary<string, HashSet<string>> InstantiatedDictionaryWithHashSet { get; set; } =
+                new Dictionary<string, HashSet<string>>();
+
+            public IDictionary<string, HashSet<string>> InstantiatedDictionaryWithHashSetWithSomeValues { get; set; } =
+                new Dictionary<string, HashSet<string>>
+                {
+                    {"item1", new HashSet<string>(new[] {"existing1", "existing2"})}
+                };
+
             public IEnumerable<string> NonInstantiatedIEnumerable { get; set; } = null!;
+            public ISet<string> InstantiatedISet { get; set; } = new HashSet<string>();
```

Comment:
Could you please also add two more test cases with concrete types (instead of `ISet<T>`) - `HashSet<T>` and `SortedSet<T>`?